src/commands/cmd_string.cc

```diff
@@ -620,6 +621,201 @@ class CommandCAD : public Commander {
+
+    // Start building the LCS table.
+    for (uint32_t i = 1; i <= alen; i++) {
+      for (uint32_t j = 1; j <= blen; j++) {
+        if (a[i - 1] == b[j - 1]) {
+          // The len LCS (and the LCS itself) of two
+          // sequences with the same final character, is the
+          // LCS of the two sequences without the last char
+          // plus that last char.
+          lcs(i, j) = lcs(i - 1, j - 1) + 1;
+        } else {
+          // If the last character is different, take the longest
+          // between the LCS of the first string and the second
+          // minus the last char, and the reverse.
+          lcs(i, j) = std::max(lcs(i - 1, j), lcs(i, j - 1));
+        }
+      }
+    }
+
+    // Store the actual LCS string if needed.
+    std::string result;
+    uint32_t idx = lcs(alen, blen);
+
+    // Allocate when we need to compute the actual LCS string.
+    bool compute_lcs = get_idx_ || !get_len_;
+    if (compute_lcs) result.resize(idx);
+
+    // Build a array if we have to emit the matched ranges.
+    std::string matches;
+    uint32_t matches_len = 0;
+
+    uint32_t i = alen;
+    uint32_t j = blen;
+    uint32_t arange_start = alen;  // alen signals that values are not set.
+    uint32_t arange_end = 0;
+    uint32_t brange_start = 0;
+    uint32_t brange_end = 0;
+    while (compute_lcs && i > 0 && j > 0) {
+      bool emit_range = false;
+      if (a[i - 1] == b[j - 1]) {
+        // If there is a match, store the character and reduce
+        // the indexes to look for a new match.
+        result.at(idx - 1) = a[i - 1];
+
+        // Track the current range.
+        if (arange_start == alen) {
+          arange_start = i - 1;
+          arange_end = i - 1;
+          brange_start = j - 1;
+          brange_end = j - 1;
+        }
+        // Let's see if we can extend the range backward since
+        // it is contiguous.
+        else if (arange_start == i && brange_start == j) {
+          arange_start--;
+          brange_start--;
+        } else {
+          emit_range = true;
+        }
+
+        // Emit the range if we matched with the first byte of
+        // one of the two strings. We'll exit the loop ASAP.
+        if (arange_start == 0 || brange_start == 0) {
+          emit_range = true;
+        }
+        idx--;
+        i--;
+        j--;
+      } else {
+        // Otherwise reduce i and j depending on the largest
+        // LCS between, to understand what direction we need to go.
+        uint32_t lcs1 = lcs(i - 1, j);
+        uint32_t lcs2 = lcs(i, j - 1);
+        if (lcs1 > lcs2)
+          i--;
+        else
+          j--;
+        if (arange_start != alen) emit_range = true;
+      }
+
+      // Emit the current range if needed.
+      uint32_t match_len = arange_end - arange_start + 1;
+      if (emit_range) {
+        if (get_idx_ && (min_match_len_ == 0 || match_len >= min_match_len_)) {
```

Comment:
sigh, don't understand why a `minmatchlen == 0` is checked here...Is this a fast path?

Reply:
Ah. When `minmatchlen == 0`, the `minmatchlen` is ignored because its default value is `0`.